birdhouse/read-configs.include.sh

```diff
@@ -302,19 +336,42 @@ process_backwards_compatible_variables() {
       new_var="${back_compat_vars#*=}"
       old_var_set="`eval "echo \\${${old_var}+set}"`"  # will equal 'set' if the variable is set, null otherwise
       if [ "${old_var_set}" = "set" ]; then
-        old_value="`eval "echo \\$${old_var}"`"
+        old_value="`eval "echo \\"\\$${old_var}\\""`"  # should keep new lines and leading empty spaces
         if [ x"${BIRDHOUSE_BACKWARD_COMPATIBLE_ALLOWED}" = x"True" ]; then
           log WARN "Deprecated variable [${old_var}] is overriding [${new_var}]. Check env.local file."
           eval 'export ${new_var}="${old_value}"'
         else
-          new_value="`eval "echo \\$${new_var}"`"
+          new_value="`eval "echo \\"\\$${new_var}\\""`"  # should keep new lines and leading empty spaces
           if [ x"${old_value}" = x"${new_value}" ]; then
             log WARN "Deprecated variable [${old_var}] can be removed as it has been superseded by [${new_var}]. Check env.local file."
           else
             log WARN "Deprecated variable [${old_var}] is present but ignored in favour of [${new_var}]. Check env.local file."
           fi
         fi
       fi
+
+      if [ x"${BIRDHOUSE_BACKWARD_COMPATIBLE_ALLOWED}" = x"True" ] && [ ! "$1" = "pre-components" ]; then
+        # Enable corresponding old var for template expansion if new var is in template expansion.
+        if echo "${VARS}" | grep -q "\s\$${new_var}\b"; then
```

Comment:
`\s` and `\b` are not very portable. I'd recommend:

```
"^[[:space:]]*\$${new_var}[[:space:]]*$"
```

(see above for other examples)

Reply:
Ah nice !